File: Fr8Data/DataTransferObjects/ContainerDTO.cs
using System;
using System.ComponentModel.DataAnnotations;
using Fr8Data.Constants;

namespace Fr8Data.DataTransferObjects
{
    public class ContainerDTO
    {
        public ContainerDTO()
        {
        }

        [Required]
        public Guid Id { get; set; }
        public string Name { get; set; }
        public Guid PlanId { get; set; }
        public int State;

        public Guid? CurrentActivityId { get; set; }
        public Guid? NextActivityId { get; set; }

        public DateTimeOffset LastUpdated { get; set; }

        public DateTimeOffset CreateDate { get; set; }

        public ActivityResponse? CurrentActivityResponse { get; set; }
        public string CurrentClientActivityName { get; set; }
        public PlanType? CurrentPlanType { get; set; }
    }
}

File: Hub/HubWeb/Scripts/app/model/ContainerDTO.ts
module dockyard.model {
    export class ContainerDTO {
        id: string;
        name: string;
        planId: number;
        state: number;
        lastUpdated: string;
        createDate: string;
        currentActivityResponse: ActivityResponse;
        currentPlanType: PlanType;
        currentClientActivityName: string;
        error: any;
        validationErrors: { [activityId: string]: ValidationResults };
    }

    export enum State {
        Unstarted = 1,
        Executing = 2,
        WaitingForTerminal = 3,
        Completed = 4,
        Failed = 5,
        Suspended = 6,
        Deleted = 7
    }

    export enum ActivityResponse {
        Null = 0,
        Success = 1,
        Error = 2,
        RequestTerminate = 3,
        RequestSuspend = 4,
        SkipChildren = 5,
        ExecuteClientAction = 7
    }

    export enum PlanType {
        OnGoing = 0,
        RunOnce = 1
    }
}